docs/random-data-loader-design.md

```diff
@@ -0,0 +1,1010 @@
+# LanceDB Random Data Loader Design Document
+
+## Overview
+
+This design document outlines the implementation of a data loader class for LanceDB in Rust that can load data from a LanceDB table in random order. The loader will be framework-agnostic and leverage a new `take` method in the `BaseTable` trait for efficient row-offset-based loading.
+
+The loader operates in two phases. First, a permutation of the base table is created. This permutation is persisted as a LanceDB table (can be persisted in memory or on disk). Next, a data loader is created from a permutation table. The data loader can be used to load data from the base table, applying the permutation.
```

Comment:
> This permutation is persisted as a LanceDB table (can be persisted in memory or on disk)

Can this be persisted in object storage? I imagine this is important if you want to do distributed training right?

Reply:
Oh, whoops...forgot I had this md still in here :laughing: 

I had originally started this PR as a vibe coding exercise before I derailed from the plan so I'm not sure how accurate it is anymore.  Let me remove it.

> Can this be persisted in object storage? I imagine this is important if you want to do distributed training right?

Yeah, it's just a table.